pkg/master/master.go

```diff
@@ -549,6 +549,46 @@ func (m *Master) InstallSwaggerAPI() {
 	swagger.RegisterSwaggerService(swaggerConfig, m.handlerContainer)
 }
 
+// Picks the best available address for an internal health check
+// TODO: Create utility function, combining this an GetHostIP in kubelet.go?
+func pickAddress(node *api.Node) string {
```

Comment:
Similar reason... It would be nice to remove the requirement that the master must have DNS cname record entries for nodes that match their k8s names (or the value passed in with --hostname_override). We have that Addresses field on the nodes which I think was added for exactly this type of thing.

I'm interested to see where #7092 goes, but as long as we have Master->Kubelet communication, I think it would be useful to get this change in. It'll make this use case cleaner, and hopefully guide new code away from using Node.Name as a resolvable address.